framework/src/interfacekernels/ADInterfaceKernel.C

```diff
@@ -0,0 +1,317 @@
+    for (_i = 0; _i < test_space.size(); _i++)
+      _local_re(_i) += raw_value(_ad_JxW[_qp] * _ad_coord[_qp] * computeQpResidual(type));
+
+  accumulateTaggedLocalResidual();
+}
+
+template <typename T>
+void
+ADInterfaceKernelTempl<T>::computeResidual()
+{
+  // in the gmsh mesh format (at least in the version 2 format) the "sideset" physical entities are
+  // associated only with the lower-dimensional geometric entity that is the boundary between two
+  // higher-dimensional element faces. It does not have a sidedness to it like the exodus format
+  // does. Consequently we may naively try to execute an interface kernel twice, one time where _var
+  // has dofs on _current_elem *AND* _neighbor_var has dofs on _neighbor_elem, and the other time
+  // where _var has dofs on _neighbor_elem and _neighbor_var has dofs on _current_elem. We only want
+  // to execute in the former case. In the future we should remove this and add some kind of "block"
+  // awareness to interface kernels to avoid all the unnecessary reinit that happens before we hit
+  // this return
+  if (!_var.activeOnSubdomain(_current_elem->subdomain_id()) ||
+      !_neighbor_var.activeOnSubdomain(_neighbor_elem->subdomain_id()))
+    return;
+
+  // Compute the residual for this element
+  computeElemNeighResidual(Moose::Element);
+
+  // Compute the residual for the neighbor
+  computeElemNeighResidual(Moose::Neighbor);
+}
+
+template <typename T>
+void
+ADInterfaceKernelTempl<T>::computeElemNeighJacobian(Moose::DGJacobianType type)
+{
+  const ADTemplateVariableTestValue<T> & test_space =
+      (type == Moose::ElementElement || type == Moose::ElementNeighbor) ? _test : _test_neighbor;
+  const ADTemplateVariableTestValue<T> & loc_phi =
+      (type == Moose::ElementElement || type == Moose::NeighborElement) ? _phi : _phi_neighbor;
+
+  unsigned int ivar, jvar;
+
+  switch (type)
+  {
+    case Moose::ElementElement:
+      ivar = jvar = _var.number();
+      resid_type = Moose::Element;
+      break;
+    case Moose::ElementNeighbor:
+      ivar = _var.number(), jvar = _neighbor_var.number();
+      resid_type = Moose::Element;
+      break;
+    case Moose::NeighborElement:
+      ivar = _neighbor_var.number(), jvar = _var.number();
+      resid_type = Moose::Neighbor;
+      break;
+    case Moose::NeighborNeighbor:
+      ivar = _neighbor_var.number(), jvar = _neighbor_var.number();
+      resid_type = Moose::Neighbor;
+      break;
+    default:
+      mooseError("Unknown DGJacobianType ", type);
+  }
+
+  if (type == Moose::ElementElement)
+    prepareMatrixTag(_assembly, ivar, jvar);
+  else
+    prepareMatrixTagNeighbor(_assembly, ivar, jvar, type);
+
+  std::size_t ad_offset;
+  if (type == Moose::ElementElement || type == Moose::NeighborElement)
+    ad_offset = jvar * _sys.getMaxVarNDofsPerElem();
+  else
+    ad_offset = jvar * _sys.getMaxVarNDofsPerElem() +
+                (_sys.system().n_vars() * _sys.getMaxVarNDofsPerElem());
```

Comment:
`ad_offset` has to match what you are doing seed? If so, it may be a good idea to create an API somewhere, so we can retrieve the same value